nilearn/maskers/surface_masker.py

```diff
@@ -182,10 +182,8 @@ def _fit_mask_img(self, img):
         img = at_least_2d(img)
         mask_data = {}
         for part, v in img.data.parts.items():
-            mask_data[part] = v.astype("float32")
-            non_finite_mask = np.logical_not(np.isfinite(mask_data[part]))
-            mask_data[part][non_finite_mask] = 0
-            mask_data[part] = mask_data[part].astype("bool").all(axis=1)
+            # mask out vertices with NaN or infinite values
+            mask_data[part] = np.isfinite(v.astype("float32")).all(axis=1)
```

Comment:
not sure what happens with nifti maskers but given that maskers should throw a warning if the data to transform contains non finite values (see https://github.com/nilearn/nilearn/pull/5359), it may be good to make sure they also warn if the image used to create the mask also contain non finite values

Reply:
Yeah I saw such a warning under _BaseSurfaceMasker too:

https://github.com/nilearn/nilearn/blob/6aa56e07b4bba646075066fe9e90df3b5fc1050d/nilearn/maskers/base_masker.py#L507

Reply:
This PR is fairly small so I can add the warning here

Reply:
done!